### PR TITLE
discard updates that have been applied during pregame simulation update

### PR DESCRIPTION
**Pull request by nukeulater**

### CODE REVIEW

File: xlive/Blam/Engine/simulation/simulation.cpp [hunk 1]
```diff
@@ -37,16 +43,38 @@ void __cdecl simulation_update_before_game_hook(int8* sim_data_out)
     p_simulation_update_before_game(sim_data_out);
 }
 
+typedef void (__cdecl* t_simulation_update_pregame)();
+t_simulation_update_pregame p_simulation_update_pregame;
+
+void __cdecl simulation_update_pregame()
+{
+    // ### TODO dedi offset
+    bool unk_9178D2 = *Memory::GetAddress<bool*>(0x5178D2, 0x0);
```
Comment (Berthalamew) [inline xlive/Blam/Engine/simulation/simulation.cpp]: Server offset: 0x520B62
This bool is "simulation_paused" in the simulation globals

File: xlive/Blam/Engine/simulation/simulation_watcher.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	static c_simulation_watcher* get()
+	{
+		// ### TODO dedi offset
+		return *Memory::GetAddress<c_simulation_watcher**>(0x5178E0, 0x0);
+	}
+
+	bool __thiscall need_to_generate_updates()
+	{
+		// ### TODO dedi offset
+		return INVOKE_TYPE(0x1D4B42, 0x0, bool(__thiscall*)(c_simulation_watcher*), this);
```
Comment (Berthalamew) [inline xlive/Blam/Engine/simulation/simulation_watcher.h]: Server offset: 0x1C188C

File: xlive/Blam/Engine/simulation/simulation_watcher.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+class c_simulation_watcher
+{
+public:
+	static c_simulation_watcher* get()
+	{
+		// ### TODO dedi offset
+		return *Memory::GetAddress<c_simulation_watcher**>(0x5178E0, 0x0);
```
Comment (Berthalamew) [inline xlive/Blam/Engine/simulation/simulation_watcher.h]: Server offset: 0x520B70

File: xlive/Blam/Engine/simulation/simulation.cpp [hunk 4]
```diff
@@ -37,16 +43,38 @@ void __cdecl simulation_update_before_game_hook(int8* sim_data_out)
...
 void simulation_apply_patches()
 {
+    // ### TODO move somewhere else, network related
+    network_memory_apply_patches();
+
     simulation_event_handler_apply_patches();
     simulation_world_apply_patches();
     DETOUR_ATTACH(p_simulation_update_before_game, Memory::GetAddress<t_simulation_update_before_game>(0x1AE902, 0x1A8B5C), simulation_update_before_game_hook);
+    // ### TODO dedi offset
+    DETOUR_ATTACH(p_simulation_update_pregame, Memory::GetAddress<t_simulation_update_pregame>(0x1AE9D3, 0x0), simulation_update_pregame);
```
Comment (Berthalamew) [inline xlive/Blam/Engine/simulation/simulation.cpp]: Server offset: 0x1A8C2D

File: xlive/Blam/Engine/simulation/simulation.cpp [hunk 5]
```diff
@@ -11,6 +12,11 @@ c_simulation_world* simulation_get_world()
 	return *Memory::GetAddress<c_simulation_world**>(0x5178DC, 0x520B6C);
 }
 
+bool simulation_engine_initialized()
+{
+    return *Memory::GetAddress<bool*>(0x5178D0, 0x0);
```
Comment (Berthalamew) [inline xlive/Blam/Engine/simulation/simulation.cpp]: Server offset: 0x520B60

File: xlive/Blam/Engine/simulation/simulation_watcher.h (hunk 2 above)
Comment (nukeulater) [reply xlive/Blam/Engine/simulation/simulation_watcher.h]: thanks

**Pull request merged**